src/model/test/OutputTableSummaryReports_GTest.cpp

```diff
@@ -0,0 +1,149 @@
+  EXPECT_EQ(3, outputTableSummaryReports.numberofSummaryReports());
+  ASSERT_FALSE(outputTableSummaryReports.addSummaryReport("MadeUpSummaryReport"));
+  EXPECT_EQ(3, outputTableSummaryReports.numberofSummaryReports());
+  EXPECT_THROW(outputTableSummaryReports.getSummaryReport(3), openstudio::Exception);
+  ASSERT_TRUE(outputTableSummaryReports.getSummaryReport(2));
+  EXPECT_EQ("SourceEnergyEndUseComponentsSummary", outputTableSummaryReports.getSummaryReport(2).get());
+  ASSERT_FALSE(outputTableSummaryReports.summaryReportIndex("AnotherMadeUpSummaryReport"));
+  ASSERT_TRUE(outputTableSummaryReports.summaryReportIndex("SourceEnergyEndUseComponentsSummary"));
+  EXPECT_EQ(2, outputTableSummaryReports.summaryReportIndex("SourceEnergyEndUseComponentsSummary").get());
+
+  outputTableSummaryReports.removeSummaryReport(1);
+  EXPECT_EQ(2, outputTableSummaryReports.numberofSummaryReports());
+  
+  // check that remaining reports moved correctly
+  std::vector<std::string> summaryReports = outputTableSummaryReports.summaryReports();
+  EXPECT_EQ("AnnualBuildingUtilityPerformanceSummary", summaryReports[0]);
+  EXPECT_EQ("SourceEnergyEndUseComponentsSummary", summaryReports[1]);
+  
+  // more remove checking
+  outputTableSummaryReports.removeAllSummaryReports();
+  EXPECT_EQ(0, outputTableSummaryReports.numberofSummaryReports());
+  outputTableSummaryReports.removeSummaryReport(0);
+  EXPECT_EQ(0, outputTableSummaryReports.numberofSummaryReports());
+  
+  // check bulk-adding summary reports
+  std::vector<std::string> summaryReportsToAdd;
+  std::string summaryReport1 = "SurfaceShadowingSummary";
+  summaryReportsToAdd.push_back(summaryReport1);
+  std::string summaryReport2 = "ShadingSummary";
+  summaryReportsToAdd.push_back(summaryReport2);
+  ASSERT_TRUE(outputTableSummaryReports.addSummaryReports(summaryReportsToAdd));
+  EXPECT_EQ(2, outputTableSummaryReports.numberofSummaryReports());
+  EXPECT_EQ(2, outputTableSummaryReports.summaryReports().size());
+}
+
+// test cloning it
+TEST_F(ModelFixture, OutputTableSummaryReports_Clone)
+{
+  // create a model to use
+  Model model;
+
+  // create an output table summary reports object to use
+  OutputTableSummaryReports outputTableSummaryReports = model.getUniqueModelObject<OutputTableSummaryReports>();
+
+  // change some of the fields
+  outputTableSummaryReports.addSummaryReport("ClimaticDataSummary");
+
+  // clone it into the same model
+  OutputTableSummaryReports outputTableSummaryReportsClone = outputTableSummaryReports.clone(model).cast<OutputTableSummaryReports>();
+  ASSERT_EQ(1, outputTableSummaryReportsClone.numberofSummaryReports());
+
+  // clone it into a different model
+  Model model2;
+  OutputTableSummaryReports outputTableSummaryReportsClone2 = outputTableSummaryReports.clone(model2).cast<OutputTableSummaryReports>();
+  ASSERT_EQ(1, outputTableSummaryReportsClone2.numberofSummaryReports());
```

Comment:
This is a unique model object. What happens if `model2` already has one instantiated? (I never thought of that...)


Also, probably test if the report is "ClimaticDataSummary"